### PR TITLE
feat(vscode): Added wizard prompt to set logic app name by User 

### PR DESCRIPTION
**Pull request by samikay101**
### PR Description
---

#### What kind of change does this PR introduce?
This PR introduces both a bug fix and a feature update. It improves the handling and naming of Logic App project folders and integrates a more dynamic structure for managing workspace folders and initializing git repositories in custom Logic App projects.

---

#### What is the current behavior?
Currently, the naming convention for Logic App projects and the structure of the workspace folders are static, defaulting to "LogicApp". Additionally, the git repository initialization doesn't check if the workspace folder is already part of a git repository, leading to potential redundant initializations.

---

#### What is the new behavior (if this is a feature change)?
- **Dynamic Folder Naming**: The Logic App project folders are now named based on the user input (`context.logicAppName`). If no name is provided, it defaults to "LogicApp", if it is a custom code workspace. If it is a deafult logic app project, we throw an error. 
- **Workspace Folder Update**: In custom code Logic App projects, a 'Functions' folder is added to the workspace for better organization.
- **Git Initialization Check**: The git repository initialization process now includes checks to prevent redundant repository initializations within existing git repositories.

---

#### Does this PR introduce a breaking change?
No breaking changes are introduced. Users will find the new dynamic naming feature and folder structure more flexible, but it won't require any changes in their existing applications.

---

### Updates Summary:

- Enhanced **folder naming** and **structure** in Logic App projects based on user-defined names.
- Added **'Functions' folder** for better organization in custom code Logic App projects.
- Implemented checks for **git repository initialization** to avoid redundancy.
- Ensured the **wizard prompt step** in the Azure Wizard is more intuitive and error-proof, mandating a project name for Logic Apps.
- **Updated `.gitignore`** for better management of Azure-specific and development-specific files and folders.

---

_This PR targets some key improvements in how Logic App projects are set up and managed, focusing on flexibility, organization, and reducing the chances of redundant operations, specifically in source control management. 

[work Item 25644784](https://msazure.visualstudio.com/One/_workitems/edit/25644784)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {
   "typescript.tsdk": "node_modules\\typescript\\lib",
   "cSpell.words": ["fluentui", "reactflow"],
-  "eslint.validate": ["json"]
+  "eslint.validate": ["json"],
+  "terminal.integrated.env.windows": {
```
Comment (ccastrotrejo) [inline .vscode/settings.json:5]: Do we need to remove this?

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/SetLogicAppNameStep.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import type { IProjectWizardContext } from '@microsoft/vscode-extension';
+import * as vscode from 'vscode';
+
+export class SetLogicAppName extends AzureWizardPromptStep<IProjectWizardContext> {
+  public async prompt(context: IProjectWizardContext): Promise<void> {
+    const logicAppName = await vscode.window.showInputBox({
+      prompt: 'Enter a name for your Logic App project',
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/SetLogicAppNameStep.ts:8]: Can we localize the text that we show to user?

File: apps/vs-code-designer/src/app/commands/createNewProject/createProjectSteps/ScriptProjectCreateStep.ts [hunk 3]
```diff
@@ -73,8 +79,15 @@ obj
 appsettings.json
 local.settings.json
 __blobstorage__
+.debug
+.gitignore
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewProject/createProjectSteps/ScriptProjectCreateStep.ts:83]: Committing or not ignoring the .gitignore, as far as I know is a good practice. I don't know if we want to ignore  it but, lets think about the person that clones the project, the .gitignore will not be cloned.

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/SetLogicAppNameStep.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          return 'Project name cannot be empty';
+        }
+        return undefined;
+      },
+    });
+
+    if (logicAppName) {
+      context.logicAppName = logicAppName;
+    } else {
+      throw new Error('Logic App project name is required.');
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/SetLogicAppNameStep.ts:20]: Will this throw Error not allowing the user continue with the experience? or how do we manage the default `LogicApp` name?

File: apps/vs-code-designer/src/app/commands/generateDeploymentScripts/iacGestureHelperFunctions.ts
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/generateDeploymentScripts/iacGestureHelperFunctions.ts:1]: Lets localize the text

**Review by ccastrotrejo — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (samikay101) [reply .vscode/settings.json:5]: This was accidental. Thanks for pointing it out.

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/SetLogicAppNameStep.ts (hunk 4 above)
Comment (samikay101) [reply apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/SetLogicAppNameStep.ts:20]: I'm currently planning to maintain the error throw in the workspace project, anticipating future updates where users may set their own LogicApp names. This approach should minimize later changes. However, I'm open to alternative views and would welcome any thoughts or suggestions you might have. We can also discuss this with the PM team if needed.

**Review by ccastrotrejo — APPROVED** [review]

**Pull request merged**